### DISCUSSION TITLE
Multiple Database Schemas and CFWheels

### DISCUSSION THREAD

**Discussion by Suspense** [wheels-dev/wheels#1372] [General]
Hello! 
I want to develop a new web application where each user's data will be stored in a separate database schema.
One database, one Lucee data source, one schema for storing common data and a set of schemas with the same structure.
The new user schema will be created after user registration.
In vanilla CFML the SQL would look like this:
`SELECT * FROM user_#session.user.id#.TableOne`
i.e.
`SELECT * FROM user_1.TableOne, SELECT * FROM user_2.TableOne and so on.`
In CFWheels I can do this:
```
component extends="Model" {
	function config() {
		table("user_#session.user.id#.TableOne");
	}
}
```
But this is a bad idea, because objects are stored in Application scope.

How to properly use the database schemas in CFWheels?
Thank you!

**Comment by Suspense** [comment]
So no any ideas?

**Comment by dbelanger** [comment]
Pretty easy to see why a question during the holidays may have been missed!  I can't speak for everyone but I know I check out completely from Dec 25th until Jan 1st.

I'm not sure why you'd need a new schema for every user but I'm sure you have your reasons.  I think it's going to be difficult to achieve what you want to achieve because of the way the scopes are loaded.  The Application scope will load before the session scope and really, the specific user information is always found in the session scope.  Wheels reads in the tables **onApplicationStart** and so, even if you had a dynamic DSN for the user set in your session scope, you won't be able to use it in **onApplicationStart** and you lose the power of the wheels ORM at that point.

I've thought about this problem a bit this morning and I don't see an easy way to involve Wheels in this kind of setup.

**Reply by Suspense** [reply]
Thank you for your opinion!
I thought there was some obvious solution that I wasn't aware of.  To use separate database schemas while maintaining ORM functionality requires significant reworking of the cfwheels core.  
Therefore, this time I abandoned this way, but this thought remained with me, may be in the future I will return to it.

**Reply by mikegrogan** [reply]
At least in Oracle you can create a synonym on a table and basically make a link to it in another schema. In theory you could move user2 tables back into the user1 schema with this approach. And they'd still live in the user2 schema as well.
